tests/test_http.py

```diff
@@ -485,26 +485,27 @@ def test_media_io_base_download_custom_request_headers(self):
     download = MediaIoBaseDownload(
         fd=self.fd, request=self.request, chunksize=3)
 
-    self.assertEqual(download._headers, {'Cache-Control':'no-store'})
+    self.assertEqual(download._headers.get('Cache-Control'), 'no-store')
 
     status, done = download.next_chunk()
 
-    result = self.fd.getvalue().decode('utf-8')
+    result = json.loads(self.fd.getvalue().decode('utf-8'))
 
     # we abuse the internals of the object we're testing, pay no attention
     # to the actual bytes= values here; we are just asserting that the
     # header we added to the original request is sent up to the server
     # on each call to next_chunk
 
-    self.assertEqual(json.loads(result),
-                     {"Cache-Control": "no-store", "range": "bytes=0-3"})
+    self.assertEqual(result.get("Cache-Control"), "no-store")
+    self.assertEqual(result.get("range"), "bytes=0-3")
```

Comment:
According to the comment above, the `range` field is an object internal that has nothing to do with what is being tested here (and thus is a potential source of flakes or confusion). I'd say just get rid of that assertion. Same for the test below where you already encountered a python 2/3 difference.